FrontEnd/src/App.js

```diff
@@ -31,10 +31,9 @@ class App extends React.Component {
         StudyId: null,
         StudyName: "",
       },
-      saving: false,
-      
-      
-      jwtTest: "Result from backend",
+      // used to disable save button for a study and its pods when saving a study or pod, based on study id
+      savingStudyId: -1,
```

Comment:
I think this should be a list of studies that is saving. It seems to me that this will only result in a ui change but still work as global lock. Aka i cant have to active jobs in 2 studies at the same time. That is almost the same as a global disable all saves. I may read the code wrong as JavaScript is very loose when it comes to types.